homeassistant/components/shelly/climate.py

```diff
@@ -98,20 +98,25 @@ async def async_restore_climate_entities(
         ent_reg, config_entry.entry_id
     )
 
+    restore_entries = []
     for entry in entries:
 
         if entry.domain != CLIMATE_DOMAIN:
             continue
 
         _LOGGER.debug("Setup sleeping climate device %s", wrapper.name)
         _LOGGER.debug("Found entry %s [%s]", entry.original_name, entry.domain)
-        async_add_entities([BlockSleepingClimate(wrapper, None, None, entry)])
+
+        restore_entries.append(BlockSleepingClimate(wrapper, None, None, entry))
+
+    if restore_entries:
+        async_add_entities(restore_entries)
 
 
 class BlockSleepingClimate(
+    update_coordinator.CoordinatorEntity,
     RestoreEntity,
     ClimateEntity,
-    entity.Entity,
 ):
     """Representation of a Shelly climate device."""
 
```

Comment:
We need to pass the coordinator (`wrapper`) to the parent init method. I'd rename `wrapper` to `coordinator` since that's our standard name for that instance, also for custom coordinators.

Reply:
We used "wrapper" all over the integration, I would say we should leave it as is for consistency.

We may consider renaming it on a different PR.

Reply:
Ok. We still need to pass it to the parent though.

Reply:
Done